packages/notebook/src/searchprovider.ts

```diff
@@ -286,16 +283,8 @@ export class NotebookSearchProvider extends SearchProvider<NotebookPanel> {
    * @returns Initial value used to populate the search box.
    */
   getInitialQuery(): string {
-    const activeCell = this.widget.content.activeCell;
-    const editor = activeCell?.editor as CodeMirrorEditor | undefined;
-    if (!editor) {
-      return '';
-    }
-    const selection = editor.state.sliceDoc(
-      editor.state.selection.main.from,
-      editor.state.selection.main.to
-    );
-    return selection;
+    // Get whatever is selected in the browser window.
+    return window.getSelection()?.toString() || '';
```

Comment:
This will require a lint pass and maybe some non-null checks but I tested it works in principle.

```suggestion
    const activeElement = document.activeElement;
    const currentWindow = activeElement instanceof HTMLIFrameElement ? activeElement.contentWindow : window;
    return currentWindow.getSelection()?.toString() || '';
```

Reply:
@JasonWeill just checking back on your thoughts on this in case if you missed the suggestion

Reply:
Just following up on this one, what do you think?

Reply:
@krassowski Thanks for your patience! I tried creating an iframe, highlighting the contents, and pressing CMD+F, but in that case, I ran the browser's "Find" function — the keyboard shortcut wasn't captured within the iframe. I can test by using the Edit > Find option in the menus.

![image](https://github.com/jupyterlab/jupyterlab/assets/93281816/61ae960c-b543-4296-9711-c65ab77b03c4)

Reply:
When I made your suggested change, in both Firefox ESR and Chrome, selecting `test` in the iframe, then pressing CMD+F, caused the browser's native find dialog to appear, not JupyterLab's. In each browser, when I chose Edit > Find from the pulldown menus, the selected `test` text was not prepopulated. Looking at the debugger, the active element at runtime is the HTML `body`, not the iframe. (This could be because selecting the command from the menus changed the active element.)

Reply:
That would be issue https://github.com/jupyterlab/jupyterlab/issues/15934

Reply:
I am saying this because I saw the same behaviour when testing https://github.com/jupyterlab/jupyterlab/pull/15857 locally yesterday.

However, I can imagine that the selector for Ctrl + F keybinding might not be reaching into the iframe if you still see this after rebasing.

Reply:
I'm still seeing the same behavior after rebasing against upstream `main`.